core/src/main/java/org/mule/runtime/core/internal/value/MuleValueProviderService.java

```diff
@@ -90,15 +99,48 @@ private ValueProvider findValueProvider(Location location, String providerName)
 
     Object component = findComponent(realLocation);
 
-    if (component instanceof ComponentValueProvider) {
-      return () -> ((ComponentValueProvider) component).getValues(providerName);
+    if (component instanceof SdkComponentValueProvider) {
+      return new ValueProvider() {
```

Comment:
Wrong adapter direction and wrong adaptation place. There must be one single and unique place at which the value providers are adapted, and that's where they're instantiated. Cannot do adaptation at the point they're used. Also, the old needs to be adapted into the new. Never adapt the new into the old

Reply:
So, should mule uses the SDK value or the SDK return the legacy Value?

Reply:
not following